rayon-core/src/lib.rs

```diff
@@ -306,27 +347,30 @@ impl Configuration {
     }
 }
 
-/// Initializes the global thread pool. This initialization is
-/// **optional**.  If you do not call this function, the thread pool
-/// will be automatically initialized with the default
-/// configuration. In fact, calling `initialize` is not recommended,
-/// except for in two scenarios:
-///
-/// - You wish to change the default configuration.
-/// - You are running a benchmark, in which case initializing may
-///   yield slightly more consistent results, since the worker threads
-///   will already be ready to go even in the first iteration.  But
-///   this cost is minimal.
-///
-/// Initialization of the global thread pool happens exactly
-/// once. Once started, the configuration cannot be
-/// changed. Therefore, if you call `initialize` a second time, it
-/// will return an error. An `Ok` result indicates that this
-/// is the first initialization of the thread pool.
+impl ThreadPoolInitError {
+    fn new(kind: ErrorKind) -> ThreadPoolInitError {
+        ThreadPoolInitError { kind: kind }
+    }
+}
+
+impl Error for ThreadPoolInitError {
+    fn description(&self) -> &str {
+        match self.kind {
+            ErrorKind::GlobalPoolAlreadyInitialized => "The global thread pool has already been initialized.",
+            ErrorKind::IOError(ref e) => e.description(),
+        }
+    }
+}
+
+impl fmt::Display for ThreadPoolInitError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.description().fmt(f)
```

Comment:
`io::Error` is more informative in `Display`, so it would be nice to forward that directly.

Reply:
Like this? https://github.com/rayon-rs/rayon/pull/516/files#diff-f42c354e8416b0ad30f65243a93d84a0R369